packages/node_modules/@ciscospark/spark-core/src/lib/spark-core-plugin-mixin.js

```diff
@@ -58,15 +58,17 @@ export default function mixinSparkCorePlugins(State, config, interceptors) {
       }));
     }
 
-
-    if (constructor.prototype._definition.ready) {
-      const {fn, depList} = State.prototype._derived.ready;
-      const def = {
-        deps: depList.concat(`${name}.ready`),
-        fn
-      };
-
-      createDerivedProperty(State.prototype, 'ready', def);
+    // Only mess with the plugin's derived properties if it's an amp-state plugin
+    if (constructor.prototype._definition) {
```

Comment:
Could this be written `if (constructor.prototype._definition && constructor.prototype._definition.ready)` ?